src/libsyntax/attr.rs

```diff
@@ -92,6 +92,13 @@ pub trait AttrMetaMethods {
     /// Gets a list of inner meta items from a list MetaItem type.
     fn meta_item_list(&self) -> Option<&[P<MetaItem>]>;
 
+    /// Indicates if the attribute is a Word.
+    fn is_word(&self) -> bool;
+    /// Indicates if the attribute is a Value String.
+    fn is_value_str(&self) -> bool;
+    /// Indicates if the attribute is a Meta-Item List.
+    fn is_meta_item_list(&self) -> bool;
```

Comment:
Can't the latter 2 have default implementations? Or maybe it's not even desirable for them to exist when `value_str` and `meta_item_list` exist.